opendevin/events/action/commands.py

```diff
@@ -60,25 +50,8 @@ class IPythonRunCellAction(Action):
     thought: str = ''
     action: str = ActionType.RUN_IPYTHON
 
-    async def run(self, controller: 'AgentController') -> 'IPythonRunCellObservation':
-        # echo "import math" | execute_cli
-        # write code to a temporary file and pass it to `execute_cli` via stdin
-        tmp_filepath = os.path.join(
-            config.get(ConfigType.WORKSPACE_BASE), '.tmp', '.ipython_execution_tmp.py'
-        )
-        pathlib.Path(os.path.dirname(tmp_filepath)).mkdir(parents=True, exist_ok=True)
-        with open(tmp_filepath, 'w') as tmp_file:
-            tmp_file.write(self.code)
-
-        tmp_filepath_inside_sandbox = os.path.join(
-            config.get(ConfigType.WORKSPACE_MOUNT_PATH_IN_SANDBOX),
-            '.tmp',
-            '.ipython_execution_tmp.py',
-        )
-        obs = controller.action_manager.run_command(
-            f'execute_cli < {tmp_filepath_inside_sandbox}', background=False
-        )
-        return IPythonRunCellObservation(content=obs.content, code=self.code)
+    def runnable(self) -> bool:
+        return True
```

Comment:
Perhaps we could also simplify the code by using class attributes directly instead of properties.

Reply:
I would love that, but managing class attributes with `dataclass` and subclasses seems to be impossible. There's no way for me to add a `runnable = False` to the base class without getting a bunch of errors about default args being set before non-defaults

Reply:
Addressed by https://github.com/OpenDevin/OpenDevin/pull/1679